File: Backend/Travle.Model/Responses/UserResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// Public user projection. Never carries the password hash/salt. <see cref="Roles"/> is a list
    /// because accounts can hold several roles at once. Read paths (list + self) carry only the small
    /// <see cref="ProfileImageThumbnail"/> (§8.2 — no heavy payloads); the full <see cref="ProfileImage"/>
    /// is populated only on the admin detail read and is null everywhere else.
    /// </summary>
    public class UserResponse
    {
        public int Id { get; set; }
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string? PhoneNumber { get; set; }

        public List<string> Roles { get; set; } = new List<string>();

        public bool IsSuspended { get; set; }
        public DateTime? SuspendedAt { get; set; }
        public string? SuspensionReason { get; set; }

        public int? CityId { get; set; }
        public string? CityName { get; set; }

        /// <summary>Whether the onboarding step should still be shown (drives the client's first-run routing).</summary>
        public bool IsOnboarded { get; set; }

        /// <summary>How many times onboarding has been shown to this user (feeds the re-prompt cap).</summary>
        public int OnboardingPromptCount { get; set; }

        public byte[]? ProfileImage { get; set; }
        public string? ProfileImageContentType { get; set; }

        /// <summary>Small JPEG avatar thumbnail; the image shipped on list + self reads (§8.2).</summary>
        public byte[]? ProfileImageThumbnail { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }
    }
}

File: UI/travle_core/lib/src/models/user_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_response.g.dart';

/// Public user projection (mirrors the backend `UserResponse`). Read paths
/// (list + self / `Access/Me`) carry only [profileImageThumbnail] — a small
/// base64 JPEG — so the avatar loads fast; the full [profileImage] is populated
/// only on the admin detail read and is null everywhere else.
@JsonSerializable()
class UserResponse {
  UserResponse({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.username,
    required this.roles,
    required this.isSuspended,
    required this.isOnboarded,
    required this.createdAt,
    this.onboardingPromptCount = 0,
    this.phoneNumber,
    this.suspendedAt,
    this.suspensionReason,
    this.cityId,
    this.cityName,
    this.profileImage,
    this.profileImageContentType,
    this.profileImageThumbnail,
    this.modifiedAt,
  });

  final int id;
  final String firstName;
  final String lastName;
  final String email;
  final String username;
  final String? phoneNumber;
  final List<String> roles;
  final bool isSuspended;
  final DateTime? suspendedAt;
  final String? suspensionReason;
  final int? cityId;
  final String? cityName;
  final bool isOnboarded;
  @JsonKey(defaultValue: 0)
  final int onboardingPromptCount;
  final String? profileImage;
  final String? profileImageContentType;

  /// Small base64 JPEG avatar thumbnail; the image shipped on list + self reads.
  final String? profileImageThumbnail;
  final DateTime createdAt;
  final DateTime? modifiedAt;

  String get fullName => '$firstName $lastName';

  factory UserResponse.fromJson(Map<String, dynamic> json) =>
      _$UserResponseFromJson(json);

  Map<String, dynamic> toJson() => _$UserResponseToJson(this);
}
